integreat_cms/static/src/js/language-tabs.ts

```diff
@@ -0,0 +1,96 @@
+const renderLanguageTabs = () => {
+    const globeElement = document.getElementById("language-switcher-globe");
+    const textElement = document.getElementById("language-switcher-text");
+    const tabWrapper = document.getElementById("tab-wrapper");
+    const languageSwitcher = document.getElementById("language-switcher");
+    const languageSwitcherListElement = document.getElementById("language-switcher").parentElement;
+    const languageSwitcherList = document.getElementById("language-switcher-list");
+
+    globeElement?.classList.add("hidden");
+    textElement?.classList.remove("hidden");
+
+    if (tabWrapper) {
+        const buffer = 20;
+        const paddingInWrapper =
+            parseFloat(window.getComputedStyle(tabWrapper).paddingLeft) +
+            parseFloat(window.getComputedStyle(tabWrapper).paddingRight);
+        const availableWidth = tabWrapper.offsetWidth - paddingInWrapper - languageSwitcherListElement.offsetWidth;
+        const tabs = Array.from(tabWrapper.children).concat(Array.from(languageSwitcherList.children));
+
+        let usedSpace = 0;
+        let tabCount = 0;
+
+        tabs.forEach((tab) => {
+            const marginOfTab = parseFloat(window.getComputedStyle(tab).marginRight);
+            const anchor = tab.getElementsByTagName("a")[0];
+
+            if (
+                usedSpace + tab.clientWidth + marginOfTab + buffer > availableWidth &&
+                tab !== languageSwitcherListElement &&
+                tabCount > 0
+            ) {
+                tab.classList.remove("mr-2");
+                if (anchor) {
+                    anchor.classList.remove(
+                        "px-4",
+                        "py-[calc(0.75rem+2px)]",
+                        "bg-water-500",
+                        "rounded-t",
+                        "hover:bg-white",
+                        "hover:text-blue-500"
+                    );
+                    anchor.classList.add("px-3", "py-2", "bg-white", "whitespace-nowrap", "hover:bg-water-500");
+                }
+                languageSwitcherList.append(tab);
+            } else {
+                tab.classList.add("mr-2");
+                if (anchor) {
+                    if (anchor.closest("#language-switcher-list") === null) {
+                        anchor.classList.remove("px-3", "py-2", "bg-white", "whitespace-nowrap", "hover:bg-water-500");
+                        anchor.classList.add(
+                            "px-4",
+                            "py-[calc(0.75rem+2px)]",
+                            "bg-water-500",
+                            "rounded-t",
+                            "hover:bg-white",
+                            "hover:text-blue-500"
+                        );
```

Comment:
I'd suggest move the list of classes to toggle to variables at the start, to eliminate the risk of accidentally changing only one part in the future resulting in inconsistent states, as well as making the logic slightly more readable.

```ts
const anchorClassesWhenShown = ["px-4", "py-[calc(0.75rem+2px)]", "bg-water-500", "rounded-t", "hover:bg-white", "hover:text-blue-500"];
const anchorClassesWhenHidden = ["px-3", "py-2", "bg-white", "whitespace-nowrap", "hover:bg-water-500"];
```

```suggestion
                    anchor.classList.remove(...anchorClassesWhenShown);
                    anchor.classList.add(...anchorClassesWhenHidden);
                }
                languageSwitcherList.append(tab);
            } else {
                tab.classList.add("mr-2");
                if (anchor) {
                    if (anchor.closest("#language-switcher-list") === null) {
                        anchor.classList.remove(...anchorClassesWhenHidden);
                        anchor.classList.add(...anchorClassesWhenShown);
```